### PR TITLE
blas/gonum: add Zgemm, Zsyrk, and Zherk

### PR DESCRIPTION
**Pull request by vladimir-ch**
Also fixes a bug in Dsyrk.

Updates #185 

Please take a look.

### CODE REVIEW

File: blas/gonum/level3double.go [hunk 1]
```diff
@@ -463,17 +463,31 @@ func (Implementation) Dsyrk(ul blas.Uplo, tA blas.Transpose, n, k int, alpha flo
 			for i := 0; i < n; i++ {
 				ctmp := c[i*ldc+i : i*ldc+n]
 				atmp := a[i*lda : i*lda+k]
-				for jc, vc := range ctmp {
```
Comment (kortschak) [inline blas/gonum/level3double.go]: How did this present?

File: blas/gonum/level3cmplx128.go [hunk 2]
```diff
@@ -254,6 +254,173 @@ func (Implementation) Zgemm(tA, tB blas.Transpose, m, n, k int, alpha complex128
...
+		}
+		return
+	}
+
+	calpha := complex(alpha, 0)
+	if trans == blas.NoTrans {
+		// Form  C = alpha*A*A^H + beta*C.
+		cbeta := complex(beta, 0)
+		if uplo == blas.Upper {
+			for i := 0; i < n; i++ {
```
Comment (kortschak) [inline blas/gonum/level3cmplx128.go]: Cases for beta==0?

File: blas/gonum/level3double.go [hunk 3]
```diff
@@ -498,7 +516,11 @@ func (Implementation) Dsyrk(ul blas.Uplo, tA blas.Transpose, n, k int, alpha flo
 	}
 	for i := 0; i < n; i++ {
 		ctmp := c[i*ldc : i*ldc+i+1]
-		if beta != 0 {
```
Comment (vladimir-ch) [inline blas/gonum/level3double.go]: The actual bug was here, the comparison `beta != 0` should have been `beta != 1`. I added the 0 special case (optimization) together with a fix, and for consistency I added it also to the other paths, all lumped into one commit. I'm sorry for taking this shortcut, I can still split it.

File: blas/gonum/level3double.go (hunk 1 above)
Comment (vladimir-ch) [reply blas/gonum/level3double.go]: As I comment below, the bug was not here.

File: blas/gonum/level3cmplx128.go (hunk 2 above)
Comment (vladimir-ch) [reply blas/gonum/level3cmplx128.go]: I'll add them.

Comment (kortschak) [reply blas/gonum/level3cmplx128.go]: This is important because of NaN behaviour.

File: blas/gonum/level3double.go (hunk 3 above)
Comment (kortschak) [reply blas/gonum/level3double.go]: No need to split it, but can you add a note to the commit message explaining the issue and what was done?

File: blas/gonum/level3cmplx128.go (hunk 2 above)
Comment (vladimir-ch) [reply blas/gonum/level3cmplx128.go]: I split it anyway, I think it's clearer this way. Also added a more descriptive message to the fixing commit.

File: blas/gonum/level3double.go (hunk 3 above)
Comment (vladimir-ch) [reply blas/gonum/level3double.go]: I split it anyway, I think it's clearer this way. Also added a more descriptive message to the fixing commit.

**Pull request merged**